code/integration-tests/runtime-tests/test/tests/pablo/constantProductDex/constantProductTests.ts

```diff
@@ -1,285 +1,1888 @@
+      expect(resultWho.toString()).to.be.equal(api.createType("AccountId32", walletTrader1.publicKey).toString());
+      expect(resultBaseAsset).to.be.bignumber.equal(new BN(hardCodedPool1.baseAssetId));
+      expect(resultQuoteAsset).to.be.bignumber.equal(new BN(hardCodedPool1.quoteAssetId));
+      expect(resultBaseAmount).to.be.bignumber.equal(new BN(0));
+      expect(resultQuoteAmount).to.be.bignumber.equal(new BN(0));
     });
 
-    it("Given that users have available funds, users can swap from the pool", async function () {
-      const quotedAmount = Pica(12);
-      const result = await swapTokenPairs(api, poolId, walletId2, baseAssetId, quoteAssetId, quotedAmount);
-      // ToDo (D. Roth): Re- enable after pablo rework!
-      // expect(result.returnedQuoteAmount.toBigInt()).to.be.equal(quotedAmount);
+    it("#4.19  I can not swap 0 amounts of any asset.", async function () {
+      const {
+        data: [resultPoolId, resultWho, resultBaseAsset, resultQuoteAsset, resultBaseAmount, resultQuoteAmount]
+      } = await sendAndWaitForSuccess(
+        api,
+        walletTrader1,
+        api.events.pablo.Swapped.is,
+        api.tx.pablo.swap(hardCodedPool1.poolId, { assetId: 130, amount: 0 }, { assetId: 4, amount: 0 }, false)
+      );
+      expect(resultPoolId).to.be.bignumber.equal(new BN(hardCodedPool1.poolId.toString()));
+      expect(resultWho.toString()).to.be.equal(api.createType("AccountId32", walletTrader1.publicKey).toString());
+      expect(resultBaseAsset).to.be.bignumber.equal(new BN(hardCodedPool1.baseAssetId));
+      expect(resultQuoteAsset).to.be.bignumber.equal(new BN(hardCodedPool1.quoteAssetId));
+      expect(resultBaseAmount).to.be.bignumber.equal(new BN(0));
+      expect(resultQuoteAmount).to.be.bignumber.equal(new BN(0));
     });
-  });
 
-  describe("ConstantProductDex Fee and Other Tests", function () {
-    if (!testConfiguration.enabledTests.feeAndOtherTests.enabled) {
-      console.log("ConstantProductDex fee and other tests are being skipped...");
-      return;
-    }
-    this.timeout(2 * 60 * 1000);
+    it("#4.7  I can not buy all of the available liquidity of a pool.", async function () {
+      const baseAssetFundsCurrentlyInPoolsBeforeTx = await api.query.tokens.accounts(
+        hardCodedPool1.poolWalletAddress,
+        hardCodedPool1.baseAssetId
+      );
+
+      const exc = await sendAndWaitForSuccess(
+        api,
+        walletTrader1,
+        api.events.pablo.Swapped.is,
+        api.tx.pablo.buy(
+          hardCodedPool1.poolId,
+          130,
+          {
+            assetId: 4,
+            amount: baseAssetFundsCurrentlyInPoolsBeforeTx.free
+          },
+          false
+        )
+      ).catch(exc => exc);
+      expect(exc.toString()).to.contain("Currently buggy! Throws wrong error message.");
+    });
+
+    it("#4.8  I can not buy with the base asset being the same as the quote asset.", async function () {
```

Comment:
Here is our expectancy is this tx to fail right? But it seems it worked.

Reply:
Yes it works but a user looses the trading fee. To me this isn't a bug since it'd require more on-chain calculations to avoid which could increase the tx fee.
The wording here is definitely outdated, thanks for pointing out. Will update accordingly.